server/user_account/views.py

```diff
@@ -197,13 +131,95 @@ def remove_token(self, request):  # pylint: disable=no-self-use
                         status=status.HTTP_200_OK)
 
 
-class AccessMembers(viewsets.ModelViewSet):
-    """
-    Allows obtaining all clients and updating them
-    """
-    http_method_names = ['get']
-    serializer_class = ClientGridSerializer
-    permission_classes = [IsAuthenticated, IsSystemAdmin]
+class CustomUserView(viewsets.ModelViewSet):
+    http_method_names = ['get', 'post', 'patch']
+    fields_to_return = None
+    fields_to_save = None
+    fields_to_filter = None
```

Comment:
> I believe this is where the crux of your change lies.
> These keywords seem to be crucial to your change as they appear numerous times in the code.
> I understand you are trying to provide more flexibility to the api. I just have a few things to mention.
> For reference I know the GithubRestAPI and others do not provide any way of filtering specific fields in their requests so it is nice that you thought of a solution for this. Especially as it removes the need for multiple ever growing serializers.
> However I think it would be possible to streamline and simplify this even more.
> For one, I dont think its nice to force the inclusion of these keywords in every request.
> 
> Here is an example from the code:
> `return this.http.get(`${this.BASEURL}/user/`, {params: {organization: localStorage.getItem('organization_id'), fields_to_return: this.returnFields, fields_to_filter: filter, fields_to_exclude: excluder}});`
> 
> Basically you want to retrieve all Users, except the current one, from a specific organization.
> We are specifying the organization in the params already yet we are once again filtering it.
> We dont really need to be excluding the current user as we can do that after we receive the data. You might just be adding more unnecessary complexity to the code by including this functionality . Is it really worth it for us ? How many requests will actually need this ?
> As a suggestion, If you want to return only specific fields you can have just 1 keyword/argument named 'fields' instead of 'feilds_to_return' to pass in the request. There would then be no need for multiple specific keywords to do the job.
> 
> In a PATCH(or POST) we know we just want to update (or create) so its redundant to have to specify a 'fields_to_save' argument like this:
> `return this.http.patch(`${this.BASEURL}/user/${id}/`, {fields_to_save: password});`
> Again here I think it can simply be 'fields' instead of 'fields_to_save' if you really wanted to have a keyword.
> 
> Code wise, I find that because certain type of behaviour is specific to the request type, it can be confusing to couple these different keywords together. Even if django already abstracts away the implementations.
> Relying on a particular set of keywords also adds an extra layer of knowledge we have to be constantly aware of and remember going forward.
> Overall its really a step in the right direction but I think I can be simplified even more.

This is true, the main part of the change was to allow for flexibility with these 3 new keywords. I agree it can be simplified more but you aren't forced to use all the keywords, only those needed. You can refer to the TestTestCase but on post and patch only fields_to_save is required, and only fields_to_return is necessary if you want to return something. Could be a good test to see if it'll be alight if I ask for nothing but exclusions and filtering are not required.

So the reason for the additional organization specification doesn't have to actually do with the change made but actually how it is expected for the Permission, changing this would require a lot more changes but it isn't impossible.

As for the validity of this, it comes down to 2 issues. Refactoring what we currently have, and preventing future redundancy. In terms of serializers, currently all of them (~5) were replaced with just one massively cleaning up our serializers. In terms of views, AccessMembers and the Users views originally have the filtering and exclusions hardcoded. Both of these have now been taken care of as well as in Dominik's PR he encounters the same issue = 3 views(really the get call but in the case of AccessMembers we had a view just for the 1 method which is ridiculous) refactored into 1. As for future possibilities, the issue with ever growing serializers is also present with the views. Whenever we want a new field to retrieve or save a new serializer had to be made, if a new retrieve request (different filters or exclusions), we need to make a new view, both of these lead to alot of cluster when this can be reduced significantly with just a bit of logic thus, I would argue we do need it and are already at the point which it is necessary.

As for the extra keywords, fields_to_return is necessary because if not, we will gain additional information then needed if not specified by the backend serializers anymore. Filters and exclusions also follow this, if needed, the keywords should be included this doesn't add much extra knowledge. In terms of the post and patch requests the fields_to_save is necessary because it is possible that the returned value desired is not only composed of what was sent if this makes sense? An example would be patching only the user_name but wanting the entire object returned. If only fields_to_save is specified this will be the only field returned. I believe some testing would have to be done to verify this however because this is my intention but in the create method only the user_name for example is returned. What the correct approach here is, I'm uncertain. We could remove this functionality for the post and patch but I'm uncertain of whether this will be all we need. Having to call additional get requests after a patch or post seems a little redundant to verify the information was correctly modified. It also opens up the question, what if a multiple post or patch were to be applied, currently we have this possibility partially available, would this ever be something we would consider rather than sending separate requests for modifying a group of users? I think for having to use an extra field but given the massive increase in flexibility is a worthy tradeoff